llvm/lib/Target/Xtensa/XtensaISelDAGToDAG.cpp

```diff
@@ -125,12 +125,90 @@ FunctionPass *llvm::createXtensaISelDag(XtensaTargetMachine &TM,
 
 void XtensaDAGToDAGISel::Select(SDNode *Node) {
   SDLoc DL(Node);
+  EVT VT = Node->getValueType(0);
 
   // If we have a custom node, we already have selected!
   if (Node->isMachineOpcode()) {
     Node->setNodeId(-1);
     return;
   }
 
+  switch (Node->getOpcode()) {
+  case ISD::SHL: {
+    SDValue N0 = Node->getOperand(0);
+    SDValue N1 = Node->getOperand(1);
+    auto *C = dyn_cast<ConstantSDNode>(N1);
+    // If C is constant in range [1..31] then we can generate SLLI
```

Comment:
The constant should always be in this range.

Reply:
Please make it consistent with SRA.

Reply:
Fixed

Reply:
The comment still needs fixing.

NB
Sentences in comments should end with a period, here and elsewhere. Sorry for having to remind you.